uni/lib/view/schedule/widgets/schedule_slot.dart

```diff
@@ -113,9 +117,19 @@ class SubjectButtonWidget extends StatelessWidget {
         'UCURR_GERAL.FICHA_UC_VIEW?pv_ocorrencia_id=$occurrId';
   }
 
-  Future<void> _launchURL() async {
-    final url = toUcLink(occurrId);
-    await launchUrl(Uri.parse(url));
+  void _launchURL(BuildContext context) {
+    final courseUnits = Provider.of<ProfileProvider>(context, listen: false)
+        .profile
+        .courseUnits;
+    final correspondingCourseUnit = courseUnits
+        .where((courseUnit) => courseUnit.occurrId == occurrId)
+        .first;
```

Comment:
Consider switching to [firstWhere](https://api.flutter.dev/flutter/dart-core/Iterable/firstWhere.html) since it is guaranteed to shortcircuit the search.

Reply:
So that we have an internal implementation like:
```
for (unit of units)
    if (unit.id == id)
       return unit;
```
Instead of 
```
for (unit of units)
    if (unit.id == id)
       matching.add(unit)
return matching;
```